### PR TITLE
Api key authorize

### PR DESCRIPTION
**Pull request by lordspinach**

### CODE REVIEW

File: src/Kirel.Blazor.HttpLogger/Program.cs [hunk 1]
```diff
@@ -18,8 +18,22 @@
...
 builder.RootComponents.Add<HeadOutlet>("head::after");
 
+var defaultBaseUri = new Uri(builder.HostEnvironment.BaseAddress);
+var hostOnlyBaseStr =  $"{defaultBaseUri.Scheme}://{defaultBaseUri.Host}:{defaultBaseUri.Port}/";
+
 var identityUriStr = builder.Configuration["Services:Identity"];
 var httpLogsStrUri = builder.Configuration["Services:HttpLogsApi"];
+if (string.IsNullOrWhiteSpace(httpLogsStrUri))
+{
+    Console.WriteLine("GatewayConfig address is not set in appsettings.json");
```
Comment (Insei) [inline src/Kirel.Blazor.HttpLogger/Program.cs]: hat is GatewayConfig?

File: src/Kirel.Blazor.Logger/Program.cs [hunk 2]
```diff
@@ -18,9 +18,40 @@
 builder.RootComponents.Add<App>("#app");
 builder.RootComponents.Add<HeadOutlet>("head::after");
 
+var defaultBaseUri = new Uri(builder.HostEnvironment.BaseAddress);
+var hostOnlyBaseStr =  $"{defaultBaseUri.Scheme}://{defaultBaseUri.Host}:{defaultBaseUri.Port}/";
+
+<<<<<<< Updated upstream
```
Comment (Insei) [inline src/Kirel.Blazor.Logger/Program.cs]: ?

File: src/Kirel.Blazor.Logger/Program.cs [hunk 3]
```diff
@@ -18,9 +18,40 @@
...
 var identityUriStr = builder.Configuration["Services:Identity"];
 var httpLogsStrUri = builder.Configuration["Services:HttpLogsApi"];
```
Comment (Insei) [inline src/Kirel.Blazor.Logger/Program.cs]: rename pls to httpLogsUriStr and next line too.

File: src/Kirel.Blazor.Logger/Program.cs [hunk 4]
```diff
@@ -18,9 +18,40 @@
...
+if (string.IsNullOrWhiteSpace(identityUriStr))
+{
+    Console.WriteLine("Identity address is not set in appsettings.json");
+    identityUriStr = hostOnlyBaseStr;
+}
+if (string.IsNullOrWhiteSpace(logsMessageStrUri))
+{
+    Console.WriteLine("Authentication address is not set in appsettings.json");
+    logsMessageStrUri = hostOnlyBaseStr;
+=======
```
Comment (Insei) [inline src/Kirel.Blazor.Logger/Program.cs]: ?

File: src/Kirel.Blazor.Logger/Program.cs [hunk 5]
```diff
@@ -18,9 +18,40 @@
...
+
+var identityUriStr = builder.Configuration["Services:Identity"];
```
Comment (Insei) [inline src/Kirel.Blazor.Logger/Program.cs]: Duplicated strings

File: src/Kirel.Blazor.Logger/Program.cs [hunk 6]
```diff
@@ -18,9 +18,40 @@
...
+var httpLogsStrUri = builder.Configuration["Services:HttpLogsApi"];
+var logsMessageStrUri = builder.Configuration["Services:MessageLogsApi"];
+var gatewayConfigUriStr = builder.Configuration["Services:GatewayConfig"];
+if (string.IsNullOrWhiteSpace(gatewayConfigUriStr))
+{
+    Console.WriteLine("GatewayConfig address is not set in appsettings.json");
+    Environment.Exit(1);
+    gatewayConfigUriStr = hostOnlyBaseStr;
+>>>>>>> Stashed changes
```
Comment (Insei) [inline src/Kirel.Blazor.Logger/Program.cs]: ?

**Review by Insei — CHANGES_REQUESTED** [review]

File: src/Kirel.Blazor.HttpLogger/Program.cs (hunk 1 above)
Comment (lordspinach) [reply src/Kirel.Blazor.HttpLogger/Program.cs]: changed

File: src/Kirel.Blazor.Logger/Program.cs (hunk 2 above)
Comment (lordspinach) [reply src/Kirel.Blazor.Logger/Program.cs]: fixed

File: src/Kirel.Blazor.Logger/Program.cs (hunk 3 above)
Comment (lordspinach) [reply src/Kirel.Blazor.Logger/Program.cs]: done

File: src/Kirel.Blazor.Logger/Program.cs (hunk 4 above)
Comment (lordspinach) [reply src/Kirel.Blazor.Logger/Program.cs]: fixed

File: src/Kirel.Blazor.Logger/Program.cs (hunk 5 above)
Comment (lordspinach) [reply src/Kirel.Blazor.Logger/Program.cs]: fixed

File: src/Kirel.Blazor.Logger/Program.cs (hunk 6 above)
Comment (lordspinach) [reply src/Kirel.Blazor.Logger/Program.cs]: fixed

**Review by Insei — APPROVED** [review]

**Pull request merged**